addons/point_of_sale/static/src/js/models.js

```diff
@@ -2971,6 +3069,184 @@ export class Order extends PosModel {
 
         return receipt;
     }
+    async printChanges() {
+        const orderChange = this.changesToOrders;
+        let isPrintSuccessful = true;
+        const d = new Date();
+        let hours = "" + d.getHours();
+        hours = hours.length < 2 ? "0" + hours : hours;
+        let minutes = "" + d.getMinutes();
+        minutes = minutes.length < 2 ? "0" + minutes : minutes;
+        for (const printer of this.pos.unwatched.printers) {
+            const changes = this._getPrintingCategoriesChanges(
+                printer.config.product_categories_ids,
+                orderChange
+            );
+            if (changes["new"].length > 0 || changes["cancelled"].length > 0) {
+                const printingChanges = {
+                    new: changes["new"],
+                    cancelled: changes["cancelled"],
+                    table_name: this.pos.config.is_table_management ? this.getTable().name : false,
+                    floor_name: this.pos.config.is_table_management
+                        ? this.getTable().floor.name
+                        : false,
+                    name: this.name || "unknown order",
+                    time: {
+                        hours,
+                        minutes,
+                    },
+                };
+                const receipt = QWeb.render("OrderChangeReceipt", { changes: printingChanges });
+                const result = await printer.print_receipt(receipt);
+                if (!result.successful) {
+                    isPrintSuccessful = false;
+                }
+            }
+        }
+
+        return isPrintSuccessful;
```

Comment:
Can we get rid of isPrintSuccessful and just return result.successful?

Reply:
The print.successfull is in a loop that runs on all printers. If I return this variable within the loop and an error occurs on the first pass. The other printers will not be processed. So no I don't think that we can remove isPrintSuccessfull